openadapt/dependencies.py

```diff
@@ -0,0 +1,109 @@
+"""This file contains the code necessary to ensure the necessary system dependencies are installed."""
+import os
+import platform
+import subprocess
+
+from loguru import logger
+
+DEP_NAME_TO_SYS_TO_INSTALL_CMD = {
+    "cargo": {
+        "Darwin": {
+            "Apple": ["brew", "install", "rust"],
+            "Intel": ["brew", "install", "rust"],
+        },
+        "Windows": [
+            "curl",
+            "--proto",
+            "'=https'",
+            "--tlsv1.2",
+            "-sSf",
+            "https://sh.rustup.rs",
+            "|",
+            "sh",
+        ],
+        "Location": ".cargo/bin",
+    },
+    "cairo": {
+        "Darwin": {
+            "Apple": ["arch", "-arm64", "brew", "install cairo"],
+            "Intel": ["brew", "install", "cairo"],
+        },
+        "Windows": "",
+        "Location": "",
+    },
+}
+
+
+def ensure_dependency(name: str, is_executable: bool = True) -> str:
+    """Returns the location of the dependency."""
```

Comment:
@dianzrong We need to do a slight change in Function docstring so that is follows the Google pydocstyle :)